fetch/fetch.go

```diff
@@ -459,9 +459,6 @@ func (f *Fetch) organizeRequests(requests []RequestMessage) map[p2p.Peer][][]Req
 	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
 	peer2requests := make(map[p2p.Peer][]RequestMessage)
 	peers := f.host.GetPeers()
-	if len(peers) == 0 {
-		panic("no peers")
```

Comment:
my concern is that there must have been a codepath that caused you to add this panic when you worked on the PR.
if we simply removed here, then it is simply kicking the can down the road. can you look deeper to see why you added this in the first place?

Reply:
I added that when I added a test of connection dropping, I just wanted to be sure that the connections I was establishing in the tests were persisting, before they were dropped forcibly.

Reply:
So its just debug and can go.

Reply:
However there is this, which is called just below. Looks like we should change that as well

 https://github.com/spacemeshos/go-spacemesh/blob/e12e38f328296553664d7592f67b921be5be2e49/fetch/fetch.go#L100-L105

Reply:
so i recall there was a earlier check on len(peers) before `organizeRequests`. 
but never mind that now. i think the easiest fix is to return no peer error from `requestHashBatchFromPeers()` instead of keep going. like i said. removing panic just kicking the can down the road.

Reply:
@countvonzero I've removed the call to `log.Fatal` as well.